File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/EnemyToUI.java
/**
 * @file EnemyToUI.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representación ToUI de EnemyStarShip
 */
public class EnemyToUI {
    private final String name;
    private final float ammoPower;
    private final float shieldPower;
    
    private LootToUI loot;
    private DamageToUI damage;   
    
    /**
     * @brief Constructor con un parámetro
     * @param enemy Instancia de la clase EnemyStarShip
     */
    EnemyToUI(EnemyStarShip enemy) {
        name=enemy.getName();
        ammoPower=enemy.getAmmoPower();
        shieldPower=enemy.getShieldPower();
    
        loot=enemy.getLoot().getUIversion();
        damage = enemy.getDamage().getUIversion();
    }

    /**
     * @brief Consultor del nombre de la nave enemiga
     * @return name
     */
    public String getName() {
        return name;
    }

    /**
     * @brief Consultor del nivel de energía de disparo de la nave enemiga
     * @return ammoPower
     */
    public float getAmmoPower() {
        return ammoPower;
    }

    /**
     * @brief Consultor del nivel de energía de escudo de la nave enemiga
     * @return shieldPower
     */
    public float getShieldPower() {
        return shieldPower;
    }

    /**
     * @brief Consultor del loot de la nave enemiga
     * @return loot
     */
    public LootToUI getLoot() {
        return loot;
    }

    /**
     * @brief Consultor del daño de la nave enemiga
     * @return damage
     */
    public DamageToUI getDamage() {
        return damage;
    } 
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/EnemyToUI.rb
#encoding:utf-8

# File: EnemyToUI.rb
# Author: Profesor

module Deepspace

# Brief: Representación ToUI de EnemyStarShip
class EnemyToUI 
  
  # Consultores
  attr_reader :name, :ammoPower, :shieldPower, :loot, :damage
  
  # Brief: Constructor con un parámetro
  # Param enemy: Instancia de la clase EnemyStarShip
  def initialize(enemy) 
    @name=enemy.name
    @ammoPower=enemy.ammoPower
    @shieldPower=enemy.shieldPower
    @loot=enemy.loot.getUIversion()
    @damage = enemy.damage.getUIversion()
  end

  # Brief: Función para representar una instancia de la clase EnemyToUI
  #        en un string
  # Return: out String que representa una instancia de la clase EnemyToUI
  def to_s
    return "Name: #{@name}, " + \
        "Ammo Power: #{@ammoPower}, " + \
        "Shield Power: #{@shieldPower}\n" + \
        "\tLoot..: #{@loot}\n" + \
        "\tDamage: #{@damage}"
  end
    
end # Class

end # Module